### ISSUE TITLE
Group Binding not working

### ISSUE THREAD

**Issue by skwiggly** [Gibe/Umbraco.Community.AzureSSO#10]
Testing with Umbraco v11.2.2. Using an existing Umbraco Administrator account.

Am able to login to Umbraco backoffice, but am not in any groups (have been removed from the Admin group), and see a blank screen with no navigation menu or content tree loaded (see attached screenshot). There is also a Javascript error in console:

`Possibly unhandled rejection: The user object is invalid, the remainingAuthSeconds is required`
(in the file: umbraco-backoffice-js.js)

Have tried several variations of group bindings:
"GroupBindings": {
    "Umbraco-Test-Admins": "Administrators"
}

With Azure AD domain:
"qw-corp.com\\Umbraco-Test-Admins": "Administrators"
"qw-corp\\Umbraco-Test-Admins": "Administrators"

And with Umbraco group alias instead of name:
"Umbraco-Test-Admins": "admin"

None of these worked, always get blank backoffice, and have to re-add my account to the Administrators group. 

Questions:
- Is this package supposed to work with existing Umbraco users, or only create new accounts?
- What is the proper structure for the group bindings config (with or without domain; group name or alias)?

![Screenshot 2023-04-24 120545](https://user-images.githubusercontent.com/3579378/234079649-ff2433b0-39ed-4c7b-acb7-56a0be56ad68.png)

**Comment by stevetemple** [comment]
Hi it's probably not getting an exact match, I think some debug logging which writes out what groups the user is in might help debug these

**Comment by mikeroyle** [comment]
Did you find a solution to this @skwiggly? I'm having the same issue.

**Comment by skwiggly** [comment]
> Did you find a solution to this @skwiggly? I'm having the same issue.

No, I tried adding logging but couldn't get Umbraco to run. Tried both integrating the source code into my solution, and also building it separately and referencing the DLL, both had errors I couldn't resolve (at least not with the time I had).

**Comment by kerrod** [comment]
+1
Having the exact same issue, and have confirmed that the group names match exactly in both AzureAD and Umbraco.

**Comment by mikeroyle** [comment]
@kerrod yes, you're right.

In Azure Token configuration, I've changed the Groups claim to use Group ID: 

![edit-groups-claim-sm](https://user-images.githubusercontent.com/1257752/235880241-4381fa6e-d72d-494e-bd3f-1ceaf8b4e71d.png)

Then updated the GroupBindings and DefaultGroups in **appsettings** using the group **Object ID** and the umbraco group **alias**:

```
"GroupBindings": {
   "xxxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxx": "admin",
   "44a38651-xxxx-4c92-b1b6-51cf26ff9bab": "editor"
},
"SetGroupsOnLogin": true,
"DefaultGroups": [
   "44a38651-xxxx-4c92-b1b6-51cf26ff9bab"
]
```
When I login now, I'm sucessfully assigned the correct Umbraco user group.

**Comment by skwiggly** [comment]
Can confirm that using Group ID in Azure config works! Thanks to @mikeroyle and @kerrod for figuring this out.

**Comment by stevetemple** [comment]
OK good info, thanks @mikeroyle and @kerrod 

I'll get this as an option in the docs

**Comment by stevetemple** [comment]
Updated the docs, so closing this issue

**Comment by TheNoisyMonk** [comment]
I would like to reopen this issue since I tried both ways from the documentation. I never seem to get the correct group. Login works fine but I am always registered as the default group. Checked the ID over and over again, only when I insert "admin" in the default I am able to get that role.
